src/platforms/esp32/main/socket_driver.c

```diff
@@ -1327,9 +1328,13 @@ static void socket_consume_mailbox(Context *ctx)
     }
 }
 
-void socket_init(Context *ctx, term opts)
+Context *socket_driver_create_driver(GlobalContext *global, term opts)
```

Comment:
maybe calling this kind of functions `socket_driver_create_port` might be clearer, but not really sure here.

Reply:
I tend to agree with you on this.  I chose "driver" because that was the term we used in the past, but it would be easy enough to change the names of the functions.  Happy to make that change in this PR, and I would like to.  So let me know if you are open to it.

Reply:
Done.